src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -229,11 +243,38 @@ impl CopyToS3Uploader {
     /// Finishes any remaining in-progress upload.
     async fn flush(&mut self) -> Result<(), anyhow::Error> {
         if let Some(uploader) = self.current_file_uploader.take() {
-            uploader.finish().await?;
+            let current_file = self.current_file_path();
+            let handle =
+                mz_ore::task::spawn(|| "s3_uploader::finish", async { uploader.finish().await });
+            let CompletedUpload {
+                part_count,
+                total_bytes_uploaded,
+            } = handle.await.unwrap()?;
+            info!(
+                "finished upload: bucket {}, file {}, bytes_uploaded {}, parts_uploaded {}",
+                &self.bucket, current_file, total_bytes_uploaded, part_count
+            );
         }
         Ok(())
     }
 
+    async fn upload_buffer(&mut self) -> Result<(), anyhow::Error> {
+        assert!(!self.buf.is_empty());
+        assert!(self.current_file_uploader.is_some());
+
+        let mut uploader = self.current_file_uploader.take().unwrap();
+        let buf = std::mem::take(&mut self.buf);
```

Comment:
i'd add a todo to make `buf` a `Bytes` so it can be cheaply cloned. this might make it possible to not have to move the `uploader` with a bit more refactoring!

Definitely not worth it right now!

Reply:
ack. (I will probably ping you about this)

Reply:
Added comment for now